nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -210,6 +214,13 @@ protected DocumentModel createEntryWithoutReferences(Map<String, Object> fieldMa
 
     @Override
     protected List<String> updateEntryWithoutReferences(DocumentModel docModel) throws DirectoryException {
+
+        String id = docModel.getId();
+        if (id == null) {
+            log.error("The document can not be updated as the id is missing");
```

Comment:
Why not throwing an exception if we're in this case ? It seems to be a Bad Request case, and we shouldn't return the OK http status if update (the only action we ask server for) doesn't happen due to wrong input.

Reply:
I wanted to keep the behaviour with the SQL directories that doesn't throw an exception in that case.